main.go

```diff
@@ -444,8 +444,55 @@ func hashWith(salt, value string) string {
 	return "z" + sum[:length]
 }
 
+type blacklistItem struct {
+	pkg      string
+	typename string
+}
+
+func buildBlacklist(files []*ast.File, info *types.Info) []blacklistItem {
+	var (
+		blacklist   []blacklistItem
+		prevReflect bool
+	)
+
+	pre := func(cursor *astutil.Cursor) bool {
+		node, ok := cursor.Node().(*ast.Ident)
+		if !ok {
+			return true
+		}
+
+		obj := info.ObjectOf(node)
+
+		tmpReflect := prevReflect
+		prevReflect = false
+
+		switch x := obj.(type) {
+		case *types.Func:
+			if x.Pkg().Name() == "reflect" && (x.Name() == "TypeOf" || x.Name() == "ValueOf") {
+				prevReflect = true
+			}
+
+		case *types.TypeName:
+			if tmpReflect {
+				blacklist = append(blacklist, blacklistItem{
+					pkg:      x.Pkg().Path(),
+					typename: x.Name(),
+				})
+			}
+		}
+
+		return true
+	}
+
+	for _, file := range files {
+		astutil.Apply(file, pre, nil)
+	}
+
+	return blacklist
+}
+
 // transformGo garbles the provided Go syntax node.
-func transformGo(file *ast.File, info *types.Info) *ast.File {
+func transformGo(file *ast.File, info *types.Info, blacklist []blacklistItem) *ast.File {
```

Comment:
I'm pretty sure the blacklist could just be `[]types.Object`, then you can compare objects directly without having to do the path and name comparisons.